frontend/javascripts/admin/dataset/composition_wizard/02_upload_files.tsx

```diff
@@ -0,0 +1,192 @@
+import { FileExcelOutlined } from "@ant-design/icons";
+import { Button } from "antd";
+import Upload, { UploadChangeParam, UploadFile } from "antd/lib/upload";
+import { AsyncButton } from "components/async_clickables";
+import { readFileAsText } from "libs/read_file";
+import Toast from "libs/toast";
+import { SoftError, values } from "libs/utils";
+import _ from "lodash";
+import { Vector3 } from "oxalis/constants";
+import { parseNml } from "oxalis/model/helpers/nml_helpers";
+import React from "react";
+import { tryToFetchDatasetsByName, WizardComponentProps, WizardContext, FileList } from "./common";
+
+const EXPECTED_VALUE_COUNT_PER_CSV_LINE = 8;
+
+export default function UploadFiles({ wizardContext, setWizardContext }: WizardComponentProps) {
+  const fileList = wizardContext.fileList;
+  const handleChange = async (info: UploadChangeParam<UploadFile<any>>) => {
+    setWizardContext((oldContext) => ({
+      ...oldContext,
+      fileList: info.fileList,
+    }));
+  };
+
+  const onPrev = () => {
+    setWizardContext((oldContext) => ({
+      ...oldContext,
+      currentWizardStep: "SelectImportType",
+    }));
+  };
+  const onNext = async () => {
+    try {
+      let newContextPartial: Partial<WizardContext> | null = null;
+      if (wizardContext.composeMode === "BIG_WARP") {
+        newContextPartial = await parseBigWarpFile(fileList);
+      } else if (wizardContext.composeMode === "WK_ANNOTATIONS") {
+        newContextPartial = await parseNmlFiles(fileList);
+      } else {
+        throw new Error("Unexpected compose mode: " + wizardContext.composeMode);
+      }
+      setWizardContext((oldContext) => ({
+        ...oldContext,
+        ...newContextPartial,
+      }));
+    } catch (exception) {
+      if (exception instanceof SoftError) {
+        Toast.warning(exception.message);
+      } else {
+        Toast.error(
+          "An error occurred while importing the uploaded files. See the Browser's console for more details.",
+        );
+        console.error(exception);
+      }
+    }
+  };
+
+  return (
+    <div>
+      <p>
+        {wizardContext.composeMode === "BIG_WARP"
+          ? "Please upload one CSV file that was exported by BigWarp."
+          : "Please upload two NML files that contain landmarks that you created with WEBKNOSSOS."}
```

Comment:
At least for me it would be helpful to get more information on how the two NML files should be structured. One or multiple trees, which nodes are matched?